test/resolve/sourceComponent.test.ts

```diff
@@ -60,7 +60,7 @@ describe('SourceComponent', () => {
   it('should return correct markedForDelete status', () => {
     expect(COMPONENT.isMarkedForDelete()).to.be.false;
     try {
```

Comment:
this would be a good place to try out the other set values (destructive types, true)